ocis-pkg/sync/mutex.go

```diff
@@ -0,0 +1,74 @@
+package sync
+
+import (
+	"sync"
+)
+
+// NRWMutex works the same as RWMutex, the only difference is that it stores mutexes in a map and reuses them.
+// It's handy if you want to write-lock, write-unlock, read-lock and read-unlock for specific names only.
+type NRWMutex struct {
```

Comment:
Also, what does the `N` in `NRWMutex` stand for? :D

Reply:
it's NamedReadWriteMutex

Reply:
Ah, thanks, makes sense